### PR TITLE
Move Property Mapper from Extbase Fluid Book

### PR DESCRIPTION
**Pull request by linawolf**
Releases: main, 11.5

### CODE REVIEW

**Review by brotkrueml — CHANGES_REQUESTED** [review]
I am not sure if the PropertyMapping should go under the Controller folder. As I have no experience with property mapping by now, I assum it can also be used independend of a controller. Maybe "Model / Domain" would be better as it is about mapping of models?

**Comment by linawolf** [comment]
> I am not sure if the PropertyMapping should go under the Controller folder. As I have no experience with property mapping by now, I assume it can also be used independent of a controller. Maybe "Model / Domain" would be better as it is about mapping of models?

I think Property Mapping is one very important feature of the Extbase Controller. One of the things that makes it so useful.

File: Documentation/ExtensionArchitecture/Extbase/Reference/Controller/PropertyMapping.rst [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+
+   .. code-block:: html
+      :caption: EXT:my_extension/Classes/Controller/SomeController.php
+
+      // use \TYPO3\CMS\Extbase\Property\PropertyMapper
+
+      /**
+       * @param PropertyMapper $propertyMapper
+       */
+      public function injectPropertyMapper(PropertyMapper $propertyMapper)
```
[the remark this answers is not in the document]
Comment (linawolf) [reply Documentation/ExtensionArchitecture/Extbase/Reference/Controller/PropertyMapping.rst:36]: With constructor injection I always have the problem, that you have to show the complete constructor so there is so much noise. I think both are äquivalent.

**Comment by brotkrueml** [comment]
> I think Property Mapping is one very important feature of the Extbase Controller. One of the things that makes it so useful.

Fine, then I have to dig into and use it :-)

File: Documentation/ExtensionArchitecture/Extbase/Reference/Controller/PropertyMapping.rst [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+
+
+Allow mapping of sub-properties
+===============================
+
+It is also possible to map to subtypes. In the above example, the
+:php:`FrontendUser` has a sub-property
+of type :php:`TYPO3\CMS\Extbase\Domain\Model\FrontendUserGroup`. If you want
+to map an incoming id, you have to configure the mapper as by default
+it won't map sub properties for security reasons:
```
[the remark this answers is not in the document]
Comment (linawolf) [reply Documentation/ExtensionArchitecture/Extbase/Reference/Controller/PropertyMapping.rst]: section was removed

**Comment by brotkrueml** [comment]
I don't like the blog_example to be used here. First: The code is not found in the blog_example repository. Second: It introduces dependencies (like the AbstractController from blog_example) which are not necessary for this chapter. Maybe we should provide the code snippets via EXT:examples or just describe it with "my_extension" without generating snippets automatically. What do you think?

**Review by brotkrueml — COMMENTED** [review]
I don't like the blog_example to be used here. First: The code is not found in the blog_example repository. Second: It introduces dependencies (like the AbstractController from blog_example) which are not necessary for this chapter. Maybe we should provide the code snippets via EXT:examples or just describe it with "my_extension" without generating snippets automatically. What do you think?

I also suggest to integrate the second snippet (injection) into the first one with proper comments.

**Comment by linawolf** [comment]
The code is found in the blog_example for v12. The snippets where automatically created from them.

Taking the code-snippets from the extension gurantees that they contain valid PHP and can be testet. Also it is the same example used all over. So readers don't have to get used to another extension just for the property mapper.

Yeah, dont like that AbstractController either, I will get rid of it in one of the next steps. Then this example will automatically change as well.

**Review by brotkrueml — APPROVED** [review]

**Pull request merged**